Simplify the getting started guide (#608)
fluxcd/flux2: docs/core-concepts/index.md

# Core Concepts

These are some of the core concepts that you should be familiar with before going over the [Getting Started Guide](get-started/index.md).

## GitOps

> GitOps is a way of managing your infrastructure and applications so that whole system is described declaratively and version controlled(most likely in a Git repository), and having an automated process that ensure that the deployed environment matches the state specified in a repository.

Reviewer: ```suggestion
GitOps is a way of managing your infrastructure and applications so that whole system is described declaratively and version controlled (most likely in a Git repository), and having an automated process that ensure that the deployed environment matches the state specified in a repository.
```

In addition,

> [..] process that ensure that the deployed [..]

should either be

> [..] process that ensures that the deployed [..]

or

> [..] process to ensure that the deployed [..]

For more information, take a look at [this documentation](https://www.gitops.tech/#what-is-gitops).

## Sources

> Sources in Flux represent sources of configuration like GitRepositories and Helm Releases.
> 
> For more information, take a look at [this documentation](components/source/gitrepositories.md)

Reviewer: ```suggestion
A *Source* defines the origin of a source and the requirements to obtain
it (e.g. credentials, version selectors). For example, the latest `1.x` tag
available from a Git repository over SSH.

Sources produce an artifact that is consumed by other Flux elements to perform
actions, like applying the contents of the artifact on the cluster. A source
may be shared by multiple consumers to deduplicate configuration and/or storage.

The origin of the source is checked for changes on a defined interval, if
there is a newer version available that matches the criteria, a new artifact
is produced.

All sources are specified as Custom Resources in a Kubernetes cluster, examples
of sources are `GitRepository`, `HelmRepository` and `Bucket` resources. 
```

Author: Thanks! Do you think we need to explain what an artifact is too?

> # Core Concepts

Reviewer 2: ```suggestion
# Core Concepts

!!! note "Work in progress"
    This document is a work in progress.
```

> 

Reviewer 3: Reminder to include "This document is a work in progress."(eg. https://tekton.dev/docs/concepts)

## GitOps

GitOps is a way of managing your infrastructure and applications so that whole system is described declaratively and version controlled (most likely in a Git repository), and having an automated process that ensures that the deployed environment matches the state specified in a repository.

For more information, take a look at [this documentation](https://www.gitops.tech/#what-is-gitops).

## Sources

A *Source* defines the origin of a source and the requirements to obtain
it (e.g. credentials, version selectors). For example, the latest `1.x` tag
available from a Git repository over SSH.

Sources produce an artifact that is consumed by other Flux elements to perform
actions, like applying the contents of the artifact on the cluster. A source
may be shared by multiple consumers to deduplicate configuration and/or storage.

The origin of the source is checked for changes on a defined interval, if
there is a newer version available that matches the criteria, a new artifact
is produced.

All sources are specified as Custom Resources in a Kubernetes cluster, examples
of sources are `GitRepository`, `HelmRepository` and `Bucket` resources. 

> For more information, take a look at [this documentation](components/source/gitrepositories.md)

Reviewer: ```suggestion
For more information, take a look at the [source-controller documentation](components/source/index.md)
```

## Reconciliation

Reconcilation refers to ensuring the a given state(e.g application running in the cluster, infrasatructure) matches a desired state declaratively defined somewhere(e.g a git repository). There are various examples of these in flux e.g:

> 1. HelmRelease reconciliation: ensures the state of the Helm release matches what is defined in the resource, performs a release if this is not the case (including revision changes of a HelmChart resource).
> 2. Bucket reconciliation: downloads and archives the contents of the declared bucket on a given interval and stores this as an artifact, records the observed revision of the artifact and the artifact itself in the status of resource.
> 3. Kustomization reconciliation: ensures the state of the application deployed on a cluster matches resources contained in a git repository.s
> 
> ## Kustomization
> 
> The kustomization represents an local set of Kubernetes resources  that Flux is supposed to reconcile in the cluster. athe  reconciliation every one minute by default but this can be specified in the kustomization. If you make any changes to the cluster using `kubectl edit` or `kubectl patch`, it will be promptly reverted. You either suspend the reconciliation or push your changes to a git repository.

Reviewer 2: Since this is an overall core concepts / glossary page, I'm thinking the main reason to mention Kustomization here (unless we also want a section on Helm as well) is to disambiguate the `Kustomization CRD` from a `kustomization` file.

Because the reconciliation default time/override/suspend note applies to all delivery controllers, I moved it into the Reconciliation section (and made just a few wording change recommendations).

I also changed the number list to unordered, since this isn't a sequence.

WDYT about these suggested changes?

```suggestion
- HelmRelease reconciliation: ensures the state of the Helm release matches what is defined in the resource, performs a release if this is not the case (including revision changes of a HelmChart resource).
- Bucket reconciliation: downloads and archives the contents of the declared bucket on a given interval and stores this as an artifact, records the observed revision of the artifact and the artifact itself in the status of resource.
- [Kustomization](#kustomization-disambiguation) reconciliation: ensures the state of the application deployed on a cluster matches resources contained in a git repository.

Reconciliation is attempted every minute by default, but this can be overridden in user-defined Flux Custom Resources. If any changes are made to the cluster using `kubectl edit` or `kubectl patch`, reconciliation will promptly revert them. You can either suspend the reconciliation, or push your changes to the defined [source](#sources) and allow reconciliation to take its course.

## Kustomization Disambiguation

The Flux [Kustomize Controller](https://toolkit.fluxcd.io/components/kustomize/controller/) installs a [Kustomization CRD](https://toolkit.fluxcd.io/components/kustomize/kustomization/). These user-defined Custom Resources specify [reconciliation](#reconciliation) instructions for the controller. Among other information, this CRD requires a path to the [kustomization](https://kubectl.docs.kubernetes.io/references/kustomize/glossary/#kustomization) file within a defined [Source](#sources). The kustomization file outlines a set of Kubernetes resource definitions that Flux is supposed to reconcile in the cluster.
```

Author: I think this looks good. I don't know if the Kustomization Disambiguation has more technical details than might be needed for the core concepts path. @hiddeco what do you think?
